pkg/operator/operator.go

```diff
@@ -401,6 +412,39 @@ func New(
 	)
 	o.informerFactories = append(o.informerFactories, kubeInformersOperatorNS)
 
+	configInformers := configv1informers.NewSharedInformerFactory(configClient, 10*time.Minute)
+	missingVersion := "0.0.1-snapshot"
+	desiredVersion := missingVersion
+	if envVersion, exists := os.LookupEnv("RELEASE_VERSION"); exists {
+		desiredVersion = envVersion
+	}
+
+	// By default, this will exit(0) if the featuregates change
+	featureGateAccessor := featuregates.NewFeatureGateAccess(
+		desiredVersion, missingVersion,
+		configInformers.Config().V1().ClusterVersions(),
+		configInformers.Config().V1().FeatureGates(),
+		eventRecorder,
+	)
+	go featureGateAccessor.Run(ctx)
+	go configInformers.Start(ctx.Done())
+
+	select {
+	case <-featureGateAccessor.InitialFeatureGatesObserved():
+		featureGates, _ := featureGateAccessor.CurrentFeatureGates()
+		klog.Infof("FeatureGates initialized: %v", featureGates.KnownFeatures())
+	case <-time.After(1 * time.Minute):
+		klog.Errorf("timed out waiting for FeatureGate detection")
+		return nil, fmt.Errorf("timed out waiting for FeatureGate detection")
+	}
+
+	featureGates, err := featureGateAccessor.CurrentFeatureGates()
+	if err != nil {
+		return nil, err
+	}
+	// read featuregate read and usage to set a variable to pass to a controller
+	o.metricsServerEnabled = featureGates.Enabled(configv1.FeatureGateMetricsServer)
```

Comment:
I see newly added FeatureGate is available in https://pkg.go.dev/github.com/openshift/api@v0.0.0-20231109185848-6cd72e415ddb/config/v1 which is current module we using

Reply:
MetricsServer featuregate is not registered in FeatureGates in yesterday's nightly image so cluster-bot test was giving error on that

Reply:
I realised I needed to vendor change in CCO https://github.com/openshift/cluster-config-operator/pull/377/files

Reply:
The idea is to pass o.metricsServerEnabled to metrics server and prometheus-adapter tasks in implementation PR